### PR TITLE
sanitize block size

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
Signed-off-by: zhangjinpeng1987 <zhangjinpeng@pingcap.com>

## What have you changed? (mandatory)

Validate block-size in RocksDB column family option.

## What are the type of the changes? (mandatory)

- Improvement (change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)

Unit test.

## Does this PR affect documentation (docs) or release note? (mandatory)

No

## Does this PR affect tidb-ansible update? (mandatory)

No

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -160,6 +161,19 @@ macro_rules! cf_config {
             pub prop_keys_index_distance: u64,
             pub titan: TitanCfConfig,
         }
+
+        impl $name {
+            fn validate(&self) -> Result<(), Box<dyn Error>> {
+                if self.block_size.0 as usize > MAX_BLOCK_SIZE {
+                    return Err(format!(
+                        "invalid block-size {}, exceed max size {}",
```
Comment (lonng) [inline src/config.rs]: Maybe we should contain the `$name` in the error message to help us to diagnose easier.

File: src/config.rs [hunk 2]
```diff
@@ -53,6 +53,7 @@ const LOCKCF_MAX_MEM: usize = GB as usize;
 const RAFT_MIN_MEM: usize = 256 * MB as usize;
 const RAFT_MAX_MEM: usize = 2 * GB as usize;
 const LAST_CONFIG_FILE: &str = "last_tikv.toml";
+const MAX_BLOCK_SIZE: usize = 32 * MB as usize;
```
Comment (Connor1996) [inline src/config.rs]: if we want to enlarge `block_size` for some purpose, is this threshold too small?

Comment (siddontang) [reply src/config.rs]: IMO, I guess 32MB is too huge.

File: src/config.rs (hunk 1 above)
Comment (zhangjinpeng1987) [reply src/config.rs]: done

**Comment by zhangjinpeng1987** [comment]
@yiwu-arbug @Connor1996 PTAL again.

**Pull request merged**